File: medolylibrary/src/main/java/com/wa2c/android/medoly/library/PluginOperationCategory.java
package com.wa2c.android.medoly.library;

import android.content.Context;
import android.support.annotation.NonNull;

/**
 * Plugin operation category.
 */
public enum PluginOperationCategory {
    /** Execute. */
    OPERATION_EXECUTE       (R.string.label_plugin_operation_execute),
    /** Open media. */
    OPERATION_MEDIA_OPEN    (R.string.label_plugin_operation_media_open),
    /** Start playing. */
    OPERATION_PLAY_START    (R.string.label_plugin_operation_play_start),
    /** Stop playing. */
    OPERATION_PLAY_STOP     (R.string.label_plugin_operation_play_stop),
    /** Close media. */
    OPERATION_MEDIA_CLOSE   (R.string.label_plugin_operation_media_close),
    /** Now playing. */
    OPERATION_PLAY_NOW      (R.string.label_plugin_operation_play_now),
    /** Complete playing. */
    OPERATION_PLAY_COMPLETE (R.string.label_plugin_operation_play_complete);



    /** Constructor. */
    PluginOperationCategory(int nameId) {
        this.nameId = nameId;
        this.categoryValue = "com.wa2c.android.medoly.plugin.category." + this.name();
    }

    /** Name id */
    private int nameId;
    /** Category value. */
    private String categoryValue;

    /** Get the category name id. */
    public int getNameId() {
        return this.nameId;
    }
    /** Get the category name. */
    public String getName(@NonNull Context context) {
        return context.getString(nameId);
    }
    /** Get the category value. */
    public String getCategoryValue() {
        return categoryValue;
    }


}

File: medolylibrary/src/main/java/com/wa2c/android/medoly/library/PluginOperationCategory.kt
package com.wa2c.android.medoly.library

import android.content.Context

/**
 * Plugin operation category.
 */
enum class PluginOperationCategory constructor(val nameId: Int) {
    /** Execute.  */
    OPERATION_EXECUTE(R.string.label_plugin_operation_execute),
    /** App launch. */
    OPERATION_APP_LAUNCH(R.string.label_plugin_operation_app_launch),
    /** Open media.  */
    OPERATION_MEDIA_OPEN(R.string.label_plugin_operation_media_open),
    /** Start playing.  */
    OPERATION_PLAY_START(R.string.label_plugin_operation_play_start),
    /** Stop playing.  */
    OPERATION_PLAY_STOP(R.string.label_plugin_operation_play_stop),
    /** Close media.  */
    OPERATION_MEDIA_CLOSE(R.string.label_plugin_operation_media_close),
    /** App quit. */
    OPERATION_APP_QUIT(R.string.label_plugin_operation_app_quit),
    /** Now playing.  */
    OPERATION_PLAY_NOW(R.string.label_plugin_operation_play_now),
    /** Complete playing.  */
    OPERATION_PLAY_COMPLETE(R.string.label_plugin_operation_play_complete);



    /** Category value.  */
    val categoryValue: String = "com.wa2c.android.medoly.plugin.category." + this.name

    /** Get the category name.  */
    fun getName(context: Context): String {
        return context.getString(nameId)
    }


}
